components/Navigation/index.js

```diff
@@ -0,0 +1,139 @@
+import { useState } from "react";
+import Link from "next/link";
+import styled from "styled-components";
+
+const NavBar = styled.nav`
+  background-color: #fff;
+  box-shadow: 3px 3px 5px rgba(0, 0, 0, 0.1);
+  width: 100vw;
+  padding: 0 10px;
+`;
+
+const Sidebar = styled.div`
+  position: fixed;
+  top: 0;
+  right: 0;
+  height: 100vh;
+  width: 50vw;
+  z-index: 999;
+  background-color: #ffffff4b;
+  box-shadow: -10px 0 10px rgba(0, 0, 0, 0.1);
+  display: ${(props) => (props.visible ? "flex" : "none")};
+  flex-direction: column;
+  align-items: flex-start;
+  justify-content: flex-start;
+  list-style: none;
+  backdrop-filter: blur(10px);
+  @media (max-width: 400px) {
+    width: 50%;
+  }
+`;
+
+const Li = styled.li`
+  height: 50px;
+  &:first-child {
+    margin-right: auto;
+    font-size: 2.2rem;
+    font-weight: 600;
+  }
+`;
+
+const Li_side = styled(Li)`
+  width: 100%;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+`;
```

Comment:
Consider merging these two into one component. 
 
 Like this.  ⬇️
 
 ```js
const ListItem = styled.li`
  height: 50px;
  &:first-child {
    margin-right: auto;
    font-size: 2.2rem;
    font-weight: 600;
  }

  ${(props) =>
    props.variant === 'side' &&
    `
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
  `}
`;

// Usage

<ListItem>Normal Item</ListItem>
<ListItem variant="side">Side Item</ListItem>

 ```
 ❗️ _Please note that this code is not tested in the project, but represents the concept. When implementing this suggestion, the corresponding props must be included when calling the component._

Reply:
Will definitely try it out at a later stage 👍🏻